packages/medusa/src/api/routes/admin/products/get-product.js

```diff
@@ -0,0 +1,24 @@
+export default async (req, res) => {
+  const { id } = req.params
+
+  const productService = req.scope.resolve("productService")
+  let product = await productService.retrieve(id)
+
+  if (!product) {
```

Comment:
Not necessary - when retrieve throws a `NOT_FOUND` error the express error handler will send the correct status code